stacks-signer/src/signer.rs

```diff
@@ -1228,16 +1236,61 @@ impl Signer {
         }
     }
 
-    /// Persist state needed to ensure the signer can continue to perform
-    /// DKG and participate in signing rounds accross crashes
-    ///
-    /// # Panics
-    /// Panics if the insertion fails
-    fn save_signer_state(&self) {
+    /// Persist signer state in both SignerDB and StackerDB
+    fn save_signer_state(&mut self) -> [Result<(), PersistenceError>; 2] {
+        let rng = &mut OsRng;
+        let signerdb_result = self.save_signer_state_in_signerdb(rng);
+        let stackerdb_result = self.save_signer_state_in_stackerdb(rng);
```

Comment:
I'm not sure if the overhead is worth caring about, but both of these functions call `encrypt` along with `state_machine.signer.save()`. Maybe add `save_signer_state(encrypted)` to `SignerStateStorage`?

Reply:
Good point, the computational overhead is probably negligible but eliminate code duplication makes maintenance easier so I'll factor this out.

Reply:
Changed.